lib/Controller/PageController.php

```diff
@@ -717,34 +717,32 @@ public function createUser( $firstname, $lastname, $mail ) {
 		
 		// if user was created successfully, send him a welcome mail
 		if( $request ) {
+
+            $welcome_mail_message = $this->settings->getSetting( 'welcome_mail_message' );
+
+			// check if password reset is active
+			if( $request && $this->settings->getSetting( 'pwd_reset_url_active' ) ) {
+				$custom_reset_link = '';
+				// get the request url
+				if( !empty( $get_link = $this->settings->getSetting( 'pwd_reset_url' ) ) && !empty( $get_attr = $this->settings->getSetting( 'pwd_reset_url_attr' ) ) && !empty( $get_attr_ldap_attr = $this->settings->getSetting( 'pwd_reset_url_attr_ldap_attr' ) ) ) {
+				    $custom_pwd_reset_link = $get_link . '&' . $get_attr . '=' . $user[ $get_attr_ldap_attr ];
+				}
+				// replace tag with custom reset link
+				str_replace( $welcome_mail_message, $this->settings->getSetting( 'pwd_reset_tag' ), $custom_reset_link);
```

Comment:
Did you test this, because this isn't the right syntax for this function.
Always a good idea to check the manual: php.net/manual/de/function.str-replace.php
The right syntax would be:
<code>str_replace( $this->settings->getSetting( 'pwd_reset_tag' ), $custom_reset_link, $welcome_mail_message );</code>